hedera-node/src/main/java/com/hedera/services/context/NodeInfo.java

```diff
@@ -55,20 +65,84 @@ public boolean isZeroStake(long nodeId) {
 		}
 
 		final int index = (int)nodeId;
-		if (index < 0 || index >= numberOfNodes) {
+		if (isIndexOutOfBounds(index)) {
 			throw new IllegalArgumentException("The address book does not have a node at index " + index);
 		}
 		return isZeroStake[index];
 	}
 
-	private void readBook() {
+	/**
+	 * Convenience method to check if this node is zero-stake.
+	 */
+	public boolean isSelfZeroStake() {
+		return isZeroStake(selfId);
+	}
+
+	/**
+	 * Returns the account parsed from the address book memo corresponding
+	 * to the given node id.
+	 *
+	 * @param nodeId the id of interest
+	 * @throws IllegalArgumentException if the book did not contain the id, or was missing an account for the id
+	 */
+	public AccountID accountOf(long nodeId) {
+		if (!bookIsRead) {
+			readBook();
+		}
+
+		final int index = (int)nodeId;
+		if (isIndexOutOfBounds(index)) {
+			throw new IllegalArgumentException("No node with id " + nodeId + " was in the address book!");
+		}
+		final var account = accounts[index];
+		if (account == null) {
+			throw new IllegalArgumentException("The address book did not have an account for node id " + nodeId + "!");
+		}
+		return account;
+	}
+
+	/**
+	 * Convenience method to check if this node has an account in the address book.
+	 */
+	public boolean hasSelfAccount() {
+		try {
+			accountOf(selfId);
+			return true;
+		} catch (IllegalArgumentException ignore) {
+			return false;
+		}
+	}
+
+	/**
+	 * Convenience method to get this node's account from the address book.
+	 *
+	 * @throws IllegalArgumentException if the node did not have an account
+	 */
+	public AccountID selfAccount() {
+		return accountOf(selfId);
+	}
+
+	private boolean isIndexOutOfBounds(int index) {
+		return index < 0 || index >= numberOfNodes;
+	}
```

Comment:
nit: You can change into `isIndexOutOfBounds(final int index, final String errorMessage)` and move the `if` and `throw(errorMessage)` logic here.